modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +469,330 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+			},
+			true,
+			true,
+		},
+		{
+			"source relayer is empty string",
+			func() {},
+			false,
+			true,
+		},
+		{
+			"fee not enabled",
+			func() {
+				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
+			},
+			true,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			// open incentivized channel
+			suite.coordinator.Setup(suite.path)
+
+			// set up coins
+			coin := ibctesting.TestCoin
+			fungibleTokenPacket := transfertypes.NewFungibleTokenPacketData(
+				coin.Denom,
+				sdk.NewInt(100).Uint64(),
+				suite.chainA.SenderAccount.GetAddress().String(),
+				suite.chainB.SenderAccount.GetAddress().String(),
+			)
+			packet := channeltypes.NewPacket(
+				fungibleTokenPacket.GetBytes(),
+				suite.chainA.SenderAccount.GetSequence(),
+				suite.path.EndpointA.ChannelConfig.PortID,
+				suite.path.EndpointA.ChannelID,
+				suite.path.EndpointB.ChannelConfig.PortID,
+				suite.path.EndpointB.ChannelID,
+				clienttypes.NewHeight(0, 100),
+				0,
+			)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			// set up & escrow packet fees
+			refundAcc := suite.chainA.SenderAccount.GetAddress()
+			packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, 1)
+			identifiedFee := types.NewIdentifiedPacketFee(
+				packetId,
+				types.Fee{
+					ReceiveFee: validCoins,
+					AckFee:     validCoins2,
+					TimeoutFee: validCoins3,
+				},
+				refundAcc.String(),
+				[]string{},
+			)
+			// escrow packet fees
+			err = suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+			suite.Require().NoError(err)
+
+			// malleate test case
+			tc.malleate()
+
+			result := cbs.OnRecvPacket(suite.chainA.GetContext(), packet, suite.chainA.SenderAccount.GetAddress())
+
+			if tc.forwardRelayer && tc.feeEnabled {
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+				}
+				suite.Require().Equal(ack, result)
+			}
+			if !tc.forwardRelayer && tc.feeEnabled {
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: "",
+				}
+				suite.Require().Equal(ack, result)
+			}
+			if tc.forwardRelayer && !tc.feeEnabled {
+				ack := channeltypes.NewResultAcknowledgement([]byte{1})
+				suite.Require().Equal(ack, result)
```

Comment:
```suggestion
                       switch {
                       case !tc.feeEnabled :
				ack := channeltypes.NewResultAcknowledgement([]byte{1})
				suite.Require().Equal(ack, result)
				
			case tc.forwardRelayer:
				ack := types.IncentivizedAcknowledgement{
					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
				}
				suite.Require().Equal(ack, result)
				
			case !tc.forwardRelayer:
				ack := types.IncentivizedAcknowledgement{
					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
					ForwardRelayerAddress: "",
				}
				suite.Require().Equal(ack, result)
			
```
just a suggestion, not sure if it compiles or is gofmt'd